### ISSUE TITLE
2 questions

### ISSUE THREAD

**Issue by mplse** [MikaelUlvesjo/shelly#1]
Hej!
Thanks for providing shelly script!
Im obviously doing something wrong. Why is the switch in the case below not being changed to set to off?
And the value of "max price today" seems to be not right (at least in the log).

current date: 2024-10-29T21:36:02+0100
21:36:03
Next run: 2024-10-29T21:45:00+0100
21:36:03
Get prises from: https://www.elprisetjustnu.se/api/v1/prices/2024/10-29_SE4.json
21:36:03
Get tomorrows prises from: https://www.elprisetjustnu.se/api/v1/prices/2024/10-30_SE4.json
21:36:04
2024-10-29T21:36:02+0100: Hour: 21, current price: 1.20854 SEK/kWh, min price today: -0.03146 SEK/kWh, max price today: 0.54111
21:36:05
SEK/kWh, avg price today: 1.4998375 SEK/kWh, always on limit: 0.5 SEK/kWh, always of limit: 2 SEK/kWh
21:36:05
2024-10-29T21:36:02+0100: Hour: 21 price: 1.20854 SEK/kWh, avg price today: 1.4998375 SEK/kWh, cut of limit: 1.4998375 SEK/kWh,
21:36:05
always on limit: 0.5 SEK/kWh, setting switch: on
21:36:05
Power usage is over inUseLimit: 0 >= -1
21:36:05
No state change... ( current state: on)
21:36:05

Settings:
priceApiEndpoint: "https://www.elprisetjustnu.se/api/v1/prices/",   // see https://www.elprisetjustnu.se/elpris-api
tomorowsPricesAfter: 15
timezone: 1
daylightSaving: true
zone: "SE4"
inUseLimit: -1.0
updateTime: 900
switchId: 0
allwaysOnMaxPrice: 0.5checkNextXHours would turn it off (price without tax or other fees)
allwaysOffMinPrice: 2.0
allwaysOnHours: [{ from: 1, to: 4 }]
onOffLimit: 1.0
checkNextXHours: 0
stopAtDataEnd: true,,
invertSwitch: false
debugMode: false
switchMode:true
colorMode: false

**Comment by Bundas77** [comment]
Hi I have the same problem! Yesterday when price was 3.27sek nothing happened.


current date: 2024-10-30T08:52:31+0100
08:52:31
Next run: 2024-10-30T9:00:00+0100
08:52:31
Get prises from: https://www.elprisetjustnu.se/api/v1/prices/2024/10-30_SE4.json
08:52:31
K/kWh, always on limit: 0.5 SEK/kWh, setting switch: on
08:52:33
2024-10-30T08:52:31+0100: Hour: 9 price: 0.15697 SEK/kWh, avg price today: 0.12945708333 SEK/kWh, cut of limit: 0.06472854166 SE
08:52:33
K/kWh, always on limit: 0.5 SEK/kWh, setting switch: on
08:52:33
2024-10-30T08:52:31+0100: Hour: 10 price: 0.06708 SEK/kWh, avg price today: 0.12945708333 SEK/kWh, cut of limit: 0.06472854166 S
08:52:33
EK/kWh, always on limit: 0.5 SEK/kWh, setting switch: on
08:52:33
Power usage is over inUseLimit: 0 >= -1
08:52:33
Inverting wanted switch state to: off
08:52:33
No state change... ( current state: off)


 priceApiEndpoint: "https://www.elprisetjustnu.se/api/v1/prices/",   // see https://www.elprisetjustnu.se/elpris-api
    tomorowsPricesAfter: 15, 
    timezone: 1, 
    daylightSaving: true,
    zone: "SE4", 
    inUseLimit: -1.0, 
    updateTime: 900, 
    switchId: 1, 
    allwaysOnMaxPrice: 0.5, 
    allwaysOffMinPrice: 1.5,
    allwaysOnHours: [{ from: 00, to: 05 },{ from: 10, to: 15 },{ from: 23, to: 00 }], 
    onOffLimit: 0.5, 
    checkNextXHours: 2, 
    stopAtDataEnd: true,
    invertSwitch: true, 
    debugMode: false, 
    colorMode: false,

**Comment by MikaelUlvesjo** [comment]
Yes it looks like there is a bug related to inUseLimit when set to - 1

**Comment by MikaelUlvesjo** [comment]
I have updated the code now, please test again.
It missed a check if CONFIG.inUseLimit >= 0.0 so the script would never turn of the power if inUseLimit was set to -1

**Comment by Soviet9773Red** [comment]
Hej Mikael,
Kul att se att du också jobbar med Shelly och använder API:et från [elprisetjustnu.se](https://www.elprisetjustnu.se/) – det är verkligen en bra grund för smart styrning!

Jag har själv också arbetat med ett liknande skript. Ursprungligen byggde jag vidare på ett finskt projekt (shelly-porssisahko), men har nu anpassat det för svenska förhållanden och stöd för SE1–SE4 via det svenska API:et.

Om du är nyfiken får du gärna kika på eller testa mitt skript:
👉 https://github.com/Soviet9773Red/shelly-elprisSE

Hör gärna av dig om du har tankar eller vill utbyta idéer!

Allt gott, Alexander
